bindata/assets/alerts/kube-apiserver-webhook.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: webhooks
+  namespace: openshift-kube-apiserver
+spec:
+  groups:
+    - name: webhooks
+      rules:
+        - alert: HighAdmissionWebhookLatency
+          annotations:
+            summary: Webhook Latency is high.
+            description: >-
+              The average latency due to the {{ $labels.name }} (type="{{ $labels.type }}") webhook is over 1s.
```

Comment:
This description does not match what the expression is computing. The following expression returns true if any of the requests during the last 5 minutes took more than 1s. 
```
(sum(rate(apiserver_admission_webhook_admission_duration_seconds_sum[5m]))by(name)-sum(rate(apiserver_admission_webhook_admission_duration_seconds_bucket{le="1"}[5m]))by(name))>0
```

If you want to alert based on the average you should use the following expression:
```
sum(rate(apiserver_admission_webhook_admission_duration_seconds_sum[5m]))by(name)-sum(rate(apiserver_admission_webhook_admission_duration_seconds_count[5m]))by(name)>1000
```

But I don't think alerting on the average is good enough, maybe on the 95th / 99th percentile might be better, wdyt?

Reply:
I have reworked to check for 99th percentile.